File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int> mp;
        mp[0] = 1;
        int s = 0, result = 0;
        for (auto i: nums) {
            s += i;
            if (mp.contains(s - k)) {
                result += mp[s - k];
            }
            mp[s]++;
        }
        return result;
    }
};

File: Data Structures & Algorithms/sort-colors/submission-2.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        ptr1, ptr2 = 0, len(nums) - 1
        cur = 0
        while cur <= ptr2:
            if nums[cur] == 0:
                nums[ptr1], nums[cur] = nums[cur], nums[ptr1]
                ptr1 += 1
                cur += 1
            elif nums[cur] == 2:
                nums[ptr2], nums[cur] = nums[cur], nums[ptr2]
                ptr2 -= 1
            else:
                cur += 1 